libs/ssl-config/src/main/java/org/opensearch/common/ssl/KeyStoreUtil.java

```diff
@@ -52,13 +53,39 @@
 import java.security.PrivateKey;
 import java.security.cert.Certificate;
 import java.util.Collection;
+import java.util.List;
 import java.util.Locale;
+import java.util.Map;
+import java.util.function.Supplier;
 
 /**
  * A variety of utility methods for working with or constructing {@link KeyStore} instances.
  */
 final class KeyStoreUtil {
 
+    public static final Supplier<Boolean> inFipsMode = () -> {
```

Comment:
Why do need `Supplier` here? The value of the FIPS mode will not change, we need to capture it only once:

```java
      public static final boolean IN_FIPS_MODE = isInFipsMode();

      private static boolean isInFipsMode() {
        try {
            // Equivalent to: boolean approvedOnly = CryptoServicesRegistrar.isInApprovedOnlyMode()
            var registrarClass = Class.forName("org.bouncycastle.crypto.CryptoServicesRegistrar");
            var isApprovedOnlyMethod = registrarClass.getMethod("isInApprovedOnlyMode");
            return (Boolean) isApprovedOnlyMethod.invoke(null);
        } catch (ReflectiveOperationException e) {
            return false;
        }
    }
```